misc.go

```diff
@@ -338,6 +346,84 @@ func max(a, b int) int {
 	return b
 }
 
+// This function parses given locale string into language tag value. Passing empty
+// string as locale means reading locale value from environment.
+func getLocaleTag(localeStr string) (language.Tag, error) {
+	localeTag := language.Und
+	var err error
+
+	if localeStr == localeStrSys {
+		// read environment locale
+		localeStr, err = locale.GetLocale()
+		if err != nil {
+			return localeTag, err
+		}
+	}
+
+	localeTag, err = language.Parse(localeStr)
+	if err != nil {
+		return localeTag, fmt.Errorf("invalid locale %q: %s", localeStr, err)
+	}
+
+	return localeTag, nil
+}
+
+// This function creates new collator for given locale. Passing empty string as
+// as locale means reading locale value from environment.
+func makeCollator(localeStr string) (*collate.Collator, error) {
+	if localeStr == localeStrDisable {
+		return nil, fmt.Errorf("locale suppose to be disabled with given string")
+	}
+
+	localeTag, err := getLocaleTag(localeStr)
+	if err != nil {
+		return nil, err
+	}
+
+	collator := collate.New(localeTag)
+
+	return collator, nil
+}
+
+// This function works like `naturalLess`, except it use locale order when
+// comparing non-digit parts of names.
+func localeNaturalLess(collator *collate.Collator, s1, s2 string) bool {
```

Comment:
This looks like a copy/paste of `naturalLess`, just to be able to use `collate` to compare the strings at the end. Is it possible to use the `collate.Numeric` option to achieve the same thing?